OpenSim/Simulation/Wrap/WrapCylinder.cpp

```diff
@@ -800,3 +801,29 @@ bool WrapCylinder::_adjust_tangent_point(SimTK::Vec3& pt1,
 
     return did_adust;
 }
+
+// Implement generateDecorations by WrapCylinder to replace the previous out of place implementation 
+// in ModelVisualizer
+void WrapCylinder::generateDecorations(bool fixed, const ModelDisplayHints& hints, const SimTK::State& state,
+    SimTK::Array_<SimTK::DecorativeGeometry>& appendToThis) const 
+{
+
+    Super::generateDecorations(fixed, hints, state, appendToThis);
+    if (fixed) return;
+
+    if (hints.get_show_wrap_geometry()) {
+        const Vec3 color(SimTK::Cyan);
+        SimTK::Transform ztoy;
+        // Make transform that takes z axis to y axis due to different
+        // assumptions between DecorativeCylinder aligned with y  and
+        // WrapCylinder aligned with z
+        ztoy.updR().setRotationFromAngleAboutX(SimTK_PI / 2);
```

Comment:
This Rotation rotates Y to Z!  You get the correct effect because you post multiply by the transform.